h2o-mapreduce-generic/src/main/java/water/hadoop/h2omapper.java

```diff
@@ -243,6 +209,54 @@ private int run2(Context context) throws IOException {
     return exitStatus;
   }
 
+  private String[] makeArgs(Configuration conf, String ice_root) throws IOException {
+    List<String> argsList = new ArrayList<String>();
+
+    // Arguments set inside the mapper.
+    argsList.add("-ice_root");
+    argsList.add(ice_root);
+    argsList.add("-hdfs_skip");
+
+    // Arguments passed by the driver.
+    int argsLength = Integer.parseInt(conf.get(H2O_MAPPER_ARGS_LENGTH));
+    for (int i = 0; i < argsLength; i++) {
+      String arg = conf.get(H2O_MAPPER_ARGS_BASE + i);
+      argsList.add(arg);
+    }
+
+    // Config files passed by the driver.
+    int confLength = Integer.parseInt(conf.get(H2O_MAPPER_CONF_LENGTH));
+    for (int i = 0; i < confLength; i++) {
+      String arg = conf.get(H2O_MAPPER_CONF_ARG_BASE + i);
+      // For files which are not passed as args (i.e. SSL certs)
+      if(null != arg && !arg.isEmpty()) {
+        argsList.add(arg);
+      }
+
+      String basename = conf.get(H2O_MAPPER_CONF_BASENAME_BASE + i);
+      String fileName = ice_root + File.separator + basename;
+      String payload = conf.get(H2O_MAPPER_CONF_PAYLOAD_BASE + i);
+      byte[] byteArr = h2odriver.convertStringToByteArr(payload);
+      h2odriver.writeBinaryFile(fileName, byteArr);
+      if(null != arg && !arg.isEmpty()) {
+        argsList.add(fileName);
+      }
+
+      // Need to modify this config here as we don't know the destination dir for keys when generating it
+      if("default-security.config".equals(basename)) {
+        modifyKeyPath(fileName, ice_root);
+      }
+    }
+
+    String ipEnvVar = conf.get(H2O_IP_ENVVAR);
+    if (!StringUtils.isNullOrEmpty(ipEnvVar)) {
+      argsList.add("-ip");
+      argsList.add(System.getenv(ipEnvVar));
```

Comment:
Should we check the env var values us nit empty?

Reply:
Check added - probably to report an explicit error. Thanks.